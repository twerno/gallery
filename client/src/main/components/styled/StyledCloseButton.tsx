import styled from 'styled-components';

export default styled.button`
    width: 1.5rem;
    height: 1.5rem;
    border: none;
    margin: 1.5rem;
    background: rgba(0, 0, 0, 0.75);
    box-shadow: 0px 0px 5px 1px rgba(0, 0, 0, 0.75);
    position: relative;
    transform: scale(1.5);

    ::before, ::after {
        content: '';
        background-color: white;
        width: 0.25rem;
        height: 1.5rem;
        position: absolute;
        left: .7rem;
        top: 0px;
        box-shadow: 0px 0px 28px -3px rgba(0,0,0,0.75);
    }

    ::before {
        transform: rotate(45deg);
    } 
    
    ::after {
        transform: rotate(-45deg);
    }

`;
