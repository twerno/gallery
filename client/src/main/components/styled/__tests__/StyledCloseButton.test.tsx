import * as React from 'react';
import { componentBasicTests } from "test/componentBasicTests";
import StyledCloseButton from "../StyledCloseButton";

describe("StyledCloseButton", () => {

    componentBasicTests("StyledCloseButton", <StyledCloseButton />);
});