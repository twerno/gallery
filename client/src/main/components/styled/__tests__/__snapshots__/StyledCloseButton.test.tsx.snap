// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledCloseButton [StyledCloseButton] Component basic tests matches the snapshot 1`] = `
.c0 {
  width: 1.5rem;
  height: 1.5rem;
  border: none;
  margin: 1.5rem;
  background: rgba(0,0,0,0.75);
  box-shadow: 0px 0px 5px 1px rgba(0,0,0,0.75);
  position: relative;
  -webkit-transform: scale(1.5);
  -ms-transform: scale(1.5);
  transform: scale(1.5);
}

.c0::before,
.c0::after {
  content: '';
  background-color: white;
  width: 0.25rem;
  height: 1.5rem;
  position: absolute;
  left: .7rem;
  top: 0px;
  box-shadow: 0px 0px 28px -3px rgba(0,0,0,0.75);
}

.c0::before {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c0::after {
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

<button
  className="c0"
/>
`;
